### PR TITLE
EN-1932 generated an empty profile the first time an user profile is edited

### PR DESCRIPTION
**Pull request by zonia3000**
The `/api/userProfiles/<user>` returned 404 because when an user is created him/her profile is not created. Now a default empty profile is generated the first time the "Edit user profile" page is accessed for a given user.

### CODE REVIEW

File: engine/src/main/java/org/entando/entando/web/userprofile/ProfileController.java [hunk 1]
```diff
@@ -71,14 +82,42 @@ public void setProfileValidator(ProfileValidator profileValidator) {
     @RequestMapping(value = "/userProfiles/{username}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity<RestResponse> getUserProfile(@PathVariable String username) throws JsonProcessingException {
         logger.debug("Requested profile -> {}", username);
+        EntityDto dto;
         if (!this.getProfileValidator().existProfile(username)) {
-            throw new RestRourceNotFoundException(EntityValidator.ERRCODE_ENTITY_DOES_NOT_EXIST, "Profile", username);
+            if (userExists(username)) {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/userprofile/ProfileController.java]: Update test coverage for this branch

File: engine/src/main/java/org/entando/entando/web/userprofile/ProfileController.java [hunk 2]
```diff
@@ -71,14 +82,42 @@ public void setProfileValidator(ProfileValidator profileValidator) {
...
+            }
+        } else {
+            dto = this.getUserProfileService().getUserProfile(username);
         }
-        EntityDto dto = this.getUserProfileService().getUserProfile(username);
         logger.debug("Main Response -> {}", dto);
         return new ResponseEntity<>(new RestResponse(dto), HttpStatus.OK);
     }
 
+    private boolean userExists(String username) {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/userprofile/ProfileController.java]: Please add simple unit test coverage for these new methods.

**Comment by entando-service** [comment]
All is well

**Comment by zonia3000** [comment]
@joewhite101 the unit test class was missing for the `ProfileController`, I added it.

**Comment by entando-service** [comment]
All is well

**Pull request merged**